src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -91,6 +107,165 @@ class LspConnectionManager {
+      return
+    }
+
+    // msg to LSP
+    const {bucket, measurement} = schema
+    const payload = {bucket: bucket?.name}
+    if (measurement) {
+      payload['measurement'] = measurement
+    }
+    // TODO: finish LSP update first
+    // this.inject(ExecuteCommand.CompositionInit, payload)
+
+    // msg to monaco-editor
+    this._setCompositionBlockStyle(
+      true,
+      schema.composition?.startLine || 1,
+      schema.composition?.startLine || 6
+    )
+
+    // handlers to trigger end composition
+    this._setCompositionToggle()
+    this._setCompositionIrreversibleExit(schema)
+  }
+
+  _endComposition(schema: SchemaSelection) {
+    // msg to LSP
+    // TODO: is there any msg to the LSP on composition end?
+    // can we have it change the yield unique identifier?
+    // this.inject(ExecuteCommand.CompositionInit, {})
+
+    // msg to monaco-editor
+    this._setCompositionBlockStyle(
+      false,
+      schema.composition?.startLine || 1,
+      schema.composition?.startLine || 1
+    )
+  }
+
+  // hack for now. To trigger for example video.
+  private _turnOnWithFirstCall = true
+
+  onSchemaSessionChange(schema: SchemaSelection, sessionCb) {
+    this._callbackSetSession = sessionCb
+    const previousState = this._session
+    this._session = {...schema}
+
+    if (!previousState?.synced && schema.synced) {
+      return this._initComposition(schema)
+    }
+    if (previousState?.synced && !schema.synced) {
+      return this._endComposition(schema)
+    }
+    // TODO: all other updates
+    // will use this.inject() --> send to LSP
+
+    // hack for now. To trigger for example video.
+    if (this._turnOnWithFirstCall) {
+      this._turnOnWithFirstCall = false
+      setTimeout(() => {
+        console.error('MOCK SCHEMA INJECTION FROM USER')
+        this._toggleSchemaSync()
+      }, 5000)
+    }
+  }
+
+  onLspCompositionChange(_jsonrpcMiddlewareResponse: unknown) {
+    // TODO: middleware to read LSP response => call this method
+    // TODO: wait for successful LSP roundtrip, then use response for:
+    // startLine and endLine --> to session store
+    // synced --> to session store
+    // all updates to this._setCompositionBlockStyle()
+  }
+
+  _compositionIcon = (synced: boolean) => {
+    return {
+      linesDecorationsClassName: synced ? ICON_SYNC_ON : ICON_SYNC_OFF,
+    }
+  }
+
+  _setCompositionBlockStyle(
+    synced: boolean,
+    startLine: number,
+    endLine: number
+  ) {
+    const syncIcon = [
+      {
+        range: new MonacoTypes.Range(startLine, 1, startLine, 1),
+        options: this._compositionIcon(synced),
+      },
+    ]
+
+    const compositionBlock = synced
+      ? [
+          {
+            range: new MonacoTypes.Range(startLine, 1, endLine, 1),
+            options: {
+              isWholeLine: true,
+              inlineClassName: 'composition-block',
+            },
+          },
+        ]
+      : []
+
+    this._compositionStyle = this._editor.deltaDecorations(
```

Comment:
I have a whole writeup on why `editor.deltaDecorations()` was chosen for composition block styling. Please feel to ask for details.

FyI: assessed, tested, and rejected options were:
* LSP `textDocument/documentColor` and `textDocument/colorPresentation`
* LSP `textDocument/documentHighlight`
* `monaco.editor.colorizeModelLine()`
* `monaco.editor.colorizeElement()`
* `monaco.editor.colorize()`